Backend/app/dashboard/rider-dashboard/src/Domain/Action/RiderPlatform/Management/Merchant.hs

```diff
@@ -0,0 +1,114 @@
+postMerchantConfigOperatingCityCreate merchantShortId opCity apiTokenInfo req@Dashboard.Common.Merchant.CreateMerchantOperatingCityReq {..} = do
+  checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
+  transaction <- SharedLogic.Transaction.buildTransaction (Domain.Types.Transaction.MerchantAPI Dashboard.Common.Merchant.PostMerchantConfigOperatingCityCreateEndpoint) (Kernel.Prelude.Just APP_BACKEND_MANAGEMENT) (Kernel.Prelude.Just apiTokenInfo) Kernel.Prelude.Nothing Kernel.Prelude.Nothing (Kernel.Prelude.Just req)
+  -- update entry in dashboard
+  merchant <- SQM.findByShortId merchantShortId >>= fromMaybeM (InvalidRequest $ "Merchant not found with shortId " <> show merchantShortId)
+  geom <- getGeomFromKML req.file >>= fromMaybeM (InvalidRequest "Cannot convert KML to Geom")
+  unless (req.city `elem` merchant.supportedOperatingCities) $
+    SQM.updateSupportedOperatingCities merchantShortId (merchant.supportedOperatingCities <> [req.city])
+  SharedLogic.Transaction.withTransactionStoring transaction $ RiderPlatformClient.RiderApp.Operations.callRiderAppOperations checkedMerchantId opCity (.merchantDSL.postMerchantConfigOperatingCityCreate) Dashboard.Common.Merchant.CreateMerchantOperatingCityReqT {geom = T.pack geom, ..}
+
+postMerchantSpecialLocationUpsert :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> ApiTokenInfo -> Kernel.Prelude.Maybe ((Kernel.Types.Id.Id Lib.Types.SpecialLocation.SpecialLocation)) -> Dashboard.Common.Merchant.UpsertSpecialLocationReq -> Environment.Flow Kernel.Types.APISuccess.APISuccess)
+postMerchantSpecialLocationUpsert merchantShortId opCity apiTokenInfo specialLocationId req = do
+  checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
+  transaction <- SharedLogic.Transaction.buildTransaction (Domain.Types.Transaction.MerchantAPI Dashboard.Common.Merchant.PostMerchantSpecialLocationUpsertEndpoint) (Kernel.Prelude.Just APP_BACKEND_MANAGEMENT) (Kernel.Prelude.Just apiTokenInfo) Kernel.Prelude.Nothing Kernel.Prelude.Nothing (Kernel.Prelude.Just req)
+  SharedLogic.Transaction.withTransactionStoring transaction $ (do error "Client call .merchantDSL.postMerchantSpecialLocationUpsert for separate helperApi couldn't be autogenerated. Logic yet to be decided" checkedMerchantId specialLocationId req)
+
+deleteMerchantSpecialLocationDelete :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> ApiTokenInfo -> Kernel.Types.Id.Id Lib.Types.SpecialLocation.SpecialLocation -> Environment.Flow Kernel.Types.APISuccess.APISuccess)
```

Comment:
Don't like how types generated with long qualified names. Thinking to add short synonyms for modules at least in domain handlers, and make it configurable in dhall